jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/applications/AbstractVariantEndpoint.java

```diff
@@ -49,35 +43,46 @@
     @Inject
     protected GenericVariantService variantService;
 
+    // required for RESTEasy
+    public AbstractVariantEndpoint() {
```

Comment:
If we would change this to
```java
public abstract class AbstractVariantEndpoint <T extends Variant> {
   private final Class<T> type;

   protected abstract AbstractVariantEndpoint(Class<T> type) {
   }
```

we could remove all the `doXXX` method and we won't need to implement a custom XXX method in each subclass.